pkg/crc/preflight/preflight_checks_darwin.go

```diff
@@ -130,6 +131,35 @@ func fixMachineDriverHyperKitInstalled() error {
 	return extractOrDownloadBinary(hyperkit.MachineDriverDownloadUrl)
 }
 
+func checkEtcHostsFilePermissions() error {
+	logging.Debugf("Checking if /etc/hosts ownership/permissions need to be adjusted after crc upgrade")
+	fileinfo, err := os.Stat(hostsFile)
+	if err != nil {
+		return err
+	}
+	// Older crc releases were setting /etc/hosts permissions to 0600 and ownership to the current user
+	// This will cause issues if ownership is reset to root:wheel with permissions
+	// issue if other tools
+	if fileinfo.Mode().Perm() == 0600 {
+		return fmt.Errorf("%s permissions are not 0644", hostsFile)
+	}
+	return nil
+}
+
+func fixEtcHostsFilePermissions() error {
+	stdOut, stdErr, err := crcos.RunWithPrivilege(fmt.Sprintf("change ownership of %s", hostsFile), "chown", "root:wheel", hostsFile)
+	if err != nil {
+		return fmt.Errorf("Unable to change ownership of %s: %s %v: %s", hostsFile, stdOut, err, stdErr)
+	}
+
+	stdOut, stdErr, err = crcos.RunWithPrivilege(fmt.Sprintf("change permissions of %s", hostsFile), "chmod", "644", hostsFile)
+	if err != nil {
+		return fmt.Errorf("Unable to change permissions of %s to 0644: %s %v: %s", hostsFile, stdOut, err, stdErr)
```

Comment:
@cfergeau does the default permission on the `/etc/hosts` is `0644` for mac (like for a user who never used the CRC) ?

Reply:
I just checked on fresh machine and looks like the permission is 644 so this looks good.